### PR TITLE
Report geometric error for 3D Tiles

### PR DESCRIPTION
**Pull request by austinEng**
Adds a debug option to display geometric error for tiles. Addresses #5016

### CODE REVIEW

File: Source/Scene/Cesium3DTileset.js [hunk 1]
```diff
@@ -351,6 +354,18 @@ define([
...
+         * This property is for debugging only; it is not optimized for production use.
+         * <p>
+         * When true, draws labels to indicate the geometric error of each tile.
+         * </p>
+         *
+         * @type {Boolean}
+         * @default false
+         */
+        this.debugShowGeometricError = defaultValue(options.debugShowGeometricError, false);
+        this._geometricErrorLabels = new LabelCollection();
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileset.js]: Initialize this to `undefined` so users don't pay for features they don't use.  Likewise, when `debugShowGeometricError` is set to `false` destroy this object.

File: Source/Scene/Cesium3DTileset.js [hunk 2]
```diff
@@ -1617,6 +1634,34 @@ define([
 
         // Number of commands added by each update above
         tileset._statistics.numberOfCommands = (commandList.length - numberOfInitialCommands);
+
+        if (tileset.debugShowGeometricError) {
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileset.js]: Move the code inside this `if` block into their own function so this function flows better.  In general, each statement in a function should be roughly at the same level of abstraction.

File: Source/Scene/Cesium3DTileset.js [hunk 3]
```diff
@@ -1617,6 +1634,34 @@ define([
...
+            tileset._geometricErrorLabels.removeAll();
+            for (i = 0; i < length; ++i) {
+                tile = selectedTiles[i];
+                if (tile.selected) {
+                    var boundingVolume = tile._boundingVolume.boundingVolume;
+                    var halfAxes = boundingVolume.halfAxes;
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileset.js]: Did you also test this with region bounding volumes?

File: Source/Scene/Cesium3DTileset.js [hunk 4]
```diff
@@ -1596,6 +1611,7 @@ define([
         updatePointAndFeatureCounts(tileset, content, true, true);
     }
 
+    var scratchCartesian = new Cartesian3();
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileset.js]: Add blank line.

**Comment by pjcozzi** [comment]
Add unit tests.

File: Source/Scene/Cesium3DTileset.js (hunk 3 above)
Comment (austinEng) [inline Source/Scene/Cesium3DTileset.js]: Yes, the `boundingVolume` property of a `TileBoundingRegion` is also an `OrientedBoundingBox`

**Comment by austinEng** [comment]
@pjcozzi updated

File: Specs/Scene/Cesium3DTilesetSpec.js [hunk 5]
```diff
@@ -1098,6 +1098,61 @@ defineSuite([
...
 
+    it ('debugShowGeometricError with regions', function() {
+        // tilesetUrl has bounding regions
+        return Cesium3DTilesTester.loadTileset(scene, tilesetUrl).then(function(tileset) {
+            tileset.debugShowGeometricError = true;
+            scene.renderForSpecs();
+            expect(tileset._geometricErrorLabels).toBeDefined();
+            expect(tileset._geometricErrorLabels.length).toEqual(5);
+            expect(tileset._geometricErrorLabels._labels[0].text).toEqual("70");
+            expect(tileset._geometricErrorLabels._labels[1].text).toEqual("0");
```
Comment (lilleyse) [inline Specs/Scene/Cesium3DTilesetSpec.js]: Use single quotes.

**Comment by pjcozzi** [comment]
Thanks @austinEng @lilleyse.

@lilleyse please merge when ready.

**Comment by pjcozzi** [comment]
Whoops, I did not mean to merge.  @austinEng please open a new PR for the style fixes.

**Comment by austinEng** [comment]
@pjcozzi I believe I had updated style before you merged, so we should be good assuming those changes are fine

**Comment by lilleyse** [comment]
Yeah they look fine.

**Pull request merged**